usecases/modulecomponents/batch/batch.go

```diff
@@ -43,14 +43,17 @@ type BatchJob struct {
 	vecs       [][]float32
 	skipObject []bool
 	startTime  time.Time
+	apiKeyHash [32]byte
 }
 
 type BatchClient interface {
 	Vectorize(ctx context.Context, input []string,
 		config moduletools.ClassConfig) (*modulecomponents.VectorizationResult, *modulecomponents.RateLimits, error)
+	GetVectorizerRateLimit(ctx context.Context) *modulecomponents.RateLimits
+	GetApiKeyHash(ctx context.Context, config moduletools.ClassConfig) [32]byte
```

Comment:
Can we have Batch Clients that don't implement `Rate Limiting` feature? If so then I would make a separate interface like so:
```
type BatchClientRateLimit interface {
        BatchClient
        GetVectorizerRateLimit(ctx context.Context) *modulecomponents.RateLimits
	GetApiKeyHash(ctx context.Context, config moduletools.ClassConfig) [32]byte
}
```

Reply:
Hmmm, the rate limits are an integral part of the bash loop and it is difficult to factor that out. I think I would prefer that the modules without without a ratelimit just return "unlimited" - just like cohere with tokens

Reply:
ok, got it